[MSVS] `True` symbol collision when compiling with some system headers

Compiler: VS 2019 16.1.4

Compile arguments: `/std:c++17 /permissive- /experimental:preprocessor /wd5105`

Code:
```cpp
#define NOMINMAX
#include <Windows.h>
#include <MsHTML.h>

#include <range/v3/all.hpp>

int main()
{
    return 1;
}
```

Error code:
```
.\range/v3/view/subrange.hpp(278): error C2872: 'True': ambiguous symbol
C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um\MsHTML.h(12540): note: could be '_BoolValue True'
.\concepts/concepts.hpp(823): note: or       'const bool concepts::defs::True'
.\range/v3/view/subrange.hpp(392): note: see reference to class template instantiation 'ranges::subrange<I,S,K>' being compiled
.\range/v3/view/subrange.hpp(276): error C2872: 'True': ambiguous symbol
C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um\MsHTML.h(12540): note: could be '_BoolValue True'
.\concepts/concepts.hpp(823): note: or       'const bool concepts::defs::True'
.\range/v3/view/subrange.hpp(276): error C2676: binary '<': '_BoolValue' does not define this operator or a conversion to a type acceptable to the predefined operator
.\range/v3/view/subrange.hpp(276): error C2059: syntax error: ')'
.\range/v3/view/subrange.hpp(276): error C2143: syntax error: missing '(' before ')'
.\range/v3/view/subrange.hpp(276): error C2440: 'type cast': cannot convert from 'ranges::subrange_kind' to 'int'
.\range/v3/view/subrange.hpp(276): note: This conversion requires an explicit cast (static_cast, C-style cast or function-style cast)
.\range/v3/view/subrange.hpp(276): error C2440: '<': cannot convert from 'ranges::subrange_kind' to '_BoolValue'
.\range/v3/view/subrange.hpp(276): note: Conversion to enumeration type requires an explicit cast (static_cast, C-style cast or function-style cast)
```

--- a/include/range/v3/view/subrange.hpp
+++ b/include/range/v3/view/subrange.hpp
@@ -275,7 +275,7 @@ namespace ranges
         template<typename R>
         constexpr CPP_ctor(subrange)(R && r, size_type n)(
             requires detail::RangeConvertibleTo_<R, I, S> &&
-                True<K == subrange_kind::sized>)
+                (K == subrange_kind::sized))
           : subrange{ranges::begin(r), ranges::end(r), n}
         {
             if(RANGES_CONSTEXPR_IF((bool)SizedRange<R>))

